ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -721,13 +737,23 @@ mkSnocket state tr = Snocket { getLocalAddr
           -- accepted.
           FDUninitialised mbLocalAddr -> mask $ \unmask -> do
             (connId, bearerInfo, simOpen) <- atomically $ do
-              bearerInfo <- stepScriptSTM (nsBearerInfo state)
               localAddress <-
                 case mbLocalAddr of
                   Just addr -> return addr
                   Nothing   -> nsNextEphemeralAddr state (getAddressType remoteAddress)
               let connId = ConnectionId { localAddress, remoteAddress }
 
+              attenuationMap <- readTVar (nsAttenuationMap state)
+
+              bearerInfo <- case Map.lookup connId attenuationMap of
+                Nothing     -> do
+                  script <- stepScriptSTMTx (nsBearerInfo state)
+                  writeTVar (nsAttenuationMap state)
+                            (Map.insert connId script attenuationMap)
+                  stepScriptSTM script
```

Comment:
If we have an explicit default script, we can just step over it.

Reply:
fixed!